src/main/java/hyperledger/besu/java/rest/client/listener/SmartContractEventListener.java

```diff
@@ -0,0 +1,27 @@
+package hyperledger.besu.java.rest.client.listener;
+
+import hyperledger.besu.java.rest.client.config.EthConfig;
+import hyperledger.besu.java.rest.client.config.EthEventsProperties;
+import hyperledger.besu.java.rest.client.dto.TopicFilter;
+import hyperledger.besu.java.rest.client.filters.TopicFilterHandler;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class SmartContractEventListener implements Runnable {
+  @Autowired private EthConfig ethConfig;
+  @Autowired private TopicFilterHandler topicFilterHandler;
+  @Autowired private EthEventsProperties ethEventsProperties;
+
+  @Override
+  public void run() {
```

Comment:
Who calls this thread?

Reply:
implemented the executor service and called the method there